src/urllib3/util/ssl_.py

```diff
@@ -83,30 +93,34 @@ def inet_pton(_, host):
 # - https://hynek.me/articles/hardening-your-web-servers-ssl-ciphers/
 #
 # The general intent is:
-# - Prefer TLS 1.3 cipher suites
 # - prefer cipher suites that offer perfect forward secrecy (DHE/ECDHE),
 # - prefer ECDHE over DHE for better performance,
 # - prefer any AES-GCM and ChaCha20 over any AES-CBC for better performance and
 #   security,
 # - prefer AES-GCM over ChaCha20 because hardware-accelerated AES is common,
-# - disable NULL authentication, MD5 MACs and DSS for security reasons.
+# - disable NULL authentication, MD5 MACs, DSS, and other
+#   insecure ciphers for security reasons.
+# - NOTE: TLS 1.3 cipher suites are managed through a different interface
+#   not exposed by CPython (yet!) and are enabled by default if they're available.
 DEFAULT_CIPHERS = ':'.join([
```

Comment:
I wish there was a more... formal way of composing this list. It doesn't exactly match up with hynek's post (per the notes), and it's somewhat different from what other platforms do (quickly looked at golang/chrome/firefox) but the spirit seems similar enough.

Somewhat of a frustrating situation.

Reply:
I agree it's a confusing configuration method. 

What parts of Hynek's post don't match? I ask because I tried to base this list off of that post while adding more explicit ordering that he offered to favor ECDSA over RSA. He mentioned it in a comment if I recall that it could be done it'd just double the size of the list.

From what I've seen of Firefox, Golang, and Chrome:
- Golang has ChaCha20 above AES-GCM (Could be wrong here but it's what it looks like to me?) which I don't think is great because of how common AES hardware acceleration is available in CPUs.
- Firefox (at least on my machine) offers AES-CBC in front of ChaCha20 where AES-CBC isn't an AEAD cipher. Also offers 3DES which is much closer to being insecure than AES or ChaCha.
- Chrome is best of the three, offers 3DES for compatibility reasons I'm sure.

The only thing we're losing from these changes is dropping static curve/DH parameters and DSS which I think are both safe to drop, DSS especially because it seems like we intended to not have it available (per the comment in our cipher section) but it was not disabled and so therefore was available. (Whoopsie!)

Reply:
I might be reading the blog post wrong, but here's the comparison:

```
| hynek        | ours                  |
+--------------+-----------------------+
 ECDH+AESGCM     ECDHE+ECDSA+AESGCM
 ECDH+CHACHA20   ECDHE+ECDSA+CHACHA20
 DH+AESGCM       ECDHE+AESGCM
 ECDH+AES256     ECDHE+CHACHA20
 DH+AES256       DHE+AESGCM
 ECDH+AES128     DHE+CHACHA20
 DH+AES          ECDHE+ECDSA+AES256
 RSA+AESGCM      ECDHE+ECDSA+AES128
 RSA+AES         ECDHE+AES256
 !aNULL          ECDHE+AES128
 !MD5            DHE+AES256
 !DSS            DHE+AES
                 RSA+AESGCM
                 RSA+AES
                 !aNULL
                 !eNULL
                 !MD5
                 !DSS
```

I mean it looks reasonable based on what I understand--I don't see anything nefarious. :) I just wish we could use a list verbatim from somewhere for the sake of not having to justify decisions.

Reply:
I changed the list to:
```python
DEFAULT_CIPHERS = ':'.join([
    'ECDH+AESGCM',
    'ECDH+CHACHA20',
    'DH+AESGCM',
    'DH+CHACHA20',
    'ECDH+AES',
    'DH+AES',
    'RSA+AESGCM',
    'RSA+AES',
    '!aNULL',
    '!eNULL',
    '!MD5',
    '!DSS',
    '!aDSS',
])
```

and I'm still getting `TLS_DH_DSS_WITH_AES_128_CBC_SHA256` in the list of offered ciphers despite `!DSS` and `!aDSS` in there and I don't understand that in the slightest.

Reply:
Okay I've changed the list to be almost identical to Hynek's list. The only changes I made are:

- Force ECDHE and DHE in front of  ECDH and DH so we get perfect forward secrecy on servers that honor cipher ordering. (OpenSSL doesn't guarantee ordering of ephemeral key exchanges before static)
- Not delineate between AES256 and AES128 because security-wise they're the same and OpenSSL already does this for us.

| Hynek         | Ours           |
|---------------|----------------|
| ECDH+AESGCM   | ECDHE+AESGCM   |
| ECDH+CHACHA20 | ECDHE+CHACHA20 |
| DH+AESGCM     | DHE+AESGCM     |
| ECDH+AES256   | DHE+CHACHA20   |
| DH+AES256     | ECDH+AESGCM    |
| ECDH+AES128   | DH+AESGCM      |
| DH+AES        | ECDH+AES       |
| RSA+AESGCM    | DH+AES         |
| RSA+AES       | RSA+AESGCM     |
| !aNULL        | RSA+AES        |
| !MD5          | !aNULL         |
| !DSS          | !eNULL         |
|               | !MD5           |
|               | !DSS           |

Want to get your approval on this list @shazow so I can reimplement the cipher list for SecureTransport.

Reply:
@sethmlarson I don't have the subject matter expertise to make an informed judgement on that. :)

As before, I don't see anything nefarious. Just asking questions so we can better document our thought process for context, in case anyone else wants to audit this later.